extensions/tasks-parent/tasks-client/src/main/java/io/resys/thena/tasks/client/spi/visitors/TaskCommandVisitor.java

```diff
@@ -111,28 +137,129 @@ private Task visitCreateTask(CreateTask command) {
     return this.current;
   }
   
-  
   private Task visitChangeTaskStatus(ChangeTaskStatus command) {
     this.current = this.current
         .withStatus(command.getStatus())
-        .withUpdated(command.getTargetDate());
+        .withUpdated(handleTargetDate(command.getTargetDate()));
     return this.current;
   }
   
   private Task visitChangeTaskPriority(ChangeTaskPriority command) {
     this.current = this.current
         .withPriority(command.getPriority())
-        .withUpdated(command.getTargetDate());
+        .withUpdated(handleTargetDate(command.getTargetDate()));
     return this.current;
   }
     
   private Task visitAssignTaskReporter(AssignTaskReporter command) {
     this.current = this.current
         .withReporterId(command.getReporterId())
-        .withUpdated(command.getTargetDate());
+        .withUpdated(handleTargetDate(command.getTargetDate()));
     return this.current;
   }
-  
+
+  private Task visitArchiveTask(ArchiveTask command) {
+    final var targetDate = handleTargetDate(command.getTargetDate());
+    this.current = this.current
+        .withArchived(targetDate)
+        .withUpdated(targetDate);
+    return this.current;
+  }
+
```

Comment:
It would probably be good to split this into two again:

1. Create a comment
2. Edit a comment

In one case, there is an id, and in the other case, there is not.

You'll probably need to maintain the same order as well.

You will also need a sanity check: Whether the item was found in the list or not.

At that stage, you'll have two very similar iterations while maintaining the same order. In both, you will find and update the task child, and if it's not found, you will throw an exception. 

You will need to create a separate function that would abstract the iteration from the mutation.  (i.e. one option is to create a visitable container with a callback function for the mutation).